File: DataAccess/src/main/java/com/TitleCounter/DataAccess/dto/GameDto.java
package com.TitleCounter.DataAccess.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.persistence.Column;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.*;

import java.time.LocalDate;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class GameDto {
    private long id;

    @NotBlank
    private String title;

    @JsonProperty("image_url")
    private String imageUrl;

    @JsonProperty("link_url")
    private String link_url;

    @NotNull
    @Min(0)
    private Long time;

    @JsonProperty("date_release")
    private LocalDate dateRelease;

    @NotNull
    @Min(0)
    @JsonProperty("global_score")
    private Float globalScore;
}

File: WinFormsApplication/Dto/GameDto.cs
﻿using System.Text.Json.Serialization;

namespace hltb.Dto
{
    public class GameDto : ISearchable
    {
        public long id { get; set; }
        public string title { get; set; }

        [JsonPropertyName("image_url")]
        public string imageUrl { get; set; }

        [JsonPropertyName("link_url")]
        public string linkUrl { get; set; }

        public long time { get; set; }

        [JsonPropertyName("date_release")]
        public string dateRelease { get; set; }

        [JsonPropertyName("global_score")]
        public float globalScore { get; set; }
        public GameDto() { }
    }
}
